designs/air-gapped.md

```diff
@@ -0,0 +1,97 @@
+As a result of their independent development, their interfaces and behaviors are not in sync, which makes the user experience not ideal.
+
+This document aims to solve that, learning from the experience gained building such tools and offering a more cohesive and simple to use solution, without drastically changing the backbone of those features and how they interact with each other.
+
+### Tenets
+
+* ****Simple:**** minimal interface, minimum number of manual steps.
+
+### Goals and Objectives
+
+As a EKS-A cluster administrator I want to:
+
+* Manage clusters in an environment without Internet connection.
+* Populate my environment with the necessary dependencies without having to worry about what those dependencies are and with minimal effort.
+* Interact with EKS-A clusters in air-gapped environments in the same way I interact with any other EKS-A cluster.
+
+### Statement of Scope
+
+**In scope**
+* Download dependencies and package them.
+* Populate disconnected environments with dependencies without Internet connection.
+
+**Not in scope**
+* Setup additional infrastructure to support disconnected environments (eg. private registries).
+
+**Future scope**
+* Package dependencies selectively.
+* Populate disconnected environments with dependencies with Internet connection.
+ 
+## Current state
+In order to support disconnected environments, users currently have to:
+1. Provision an OCI registry.
+1. Run `eksctl anywhere download artifacts` to download manifests to disk.
+1. Run `eksctl anywhere download images` to download images and charts to disk.
+1. Run `eksctl anywhere import images` to push container images and charts to a registry.
+1. Configure their cluster spec file to point to the registry 
+1. Run all cluster commands (`create`, `upgrade`, etc.) with the `--bundles-override` flag pointing to the `Bundles` manifest downloaded in the first step.
+
+## Overview of Solution
+1. Users will provide an OCI registry.
+1. We will store all dependencies (container images, helm charts and yaml manifests) in that registry.
+1. If a cluster config with a [`registryMirrorConfiguration`](https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_types.go#L51) is provided, the CLI and the rest of components will always pull them from it.
+No extra flags/configuration will be needed and the user will interact with this cluster as with any other one.
+
+## Solution Details
+
+### Populating the registry
+
+In order to populate the registry, we will offer two commands:
+```sh
+eksctl anywhere export artifacts --output artifacts.tar
+eksctl anywhere import artifacts --input artifacts.tar --registry myregistry.com
+```
+
+The first one downloads all dependencies (yaml manifests, images and helm charts). The only command argument here is the destination file. This command will create a tarball containing all 3 types of artifacts in that location.
```

Comment:
will this download the artifacts for individual components (CAPI/CAPV etc) across all k8s versions that the cli release in use supports?

Reply:
Yeah, exactly